src/material_library.cpp

```diff
@@ -0,0 +1,221 @@
+#include <unistd.h>
+#include <iostream>
+
+#ifndef PYNE_IS_AMALGAMATED
+#include "material_library.h"
+#endif
+
+namespace pyne {
+// Empty Constructor
+MaterialLibrary::MaterialLibrary(){};
+
+// Default constructor
+MaterialLibrary::MaterialLibrary(std::string file, std::string datapath) {
+  if (!check_file_exists(file)) {
+    std::cerr << "The file " << file << " does not exist or is read protected"
+              << std::endl;
+    exit(1);
```

Comment:
Maybe throw here? It looks like there a couple custom PyNE exceptions in utils.h.

Reply:
Specifically the `FileNotFound` exception. There's also a `file_exists` method in there - maybe it could be replaced with your version below at some point?

Reply:
Use `FileNotFound` exception may lead to garbled text. Use `throw std::runtime_error` is also an available choice, such as the code here:
https://github.com/pyne/pyne/blob/b0be1c69d5523f99fd07bea9256bbf9c302ec9f1/src/source_sampling.cpp#L63

Reply:
Do we know why it leads to garbled text? Maybe we should fix it?

Reply:
Just took a look. The `FileNotFound::what()` returns a char* generated by a string which goes out of scope at the end of that function. That should be fixed at some point. I'll make an issue...